src/socialMediaActivity.js

```diff
@@ -82,82 +84,61 @@ function registerPlatformListener(platform, eventType, blockingType, callback) {
+ * @param {Array} listenerArguments - The event about to be sent to the listener.
+ * @param {EventOptions} options - The set of options provided when this listener was registered.
+ * @return {boolean} Whether to notify this listener for this event.
  */
-export function registerFacebookActivityTracker(
-    callback,
-    events,
-    blocking = false ){
-    if (events.includes("post") || events.includes("<all_events>")) {
-        registerPlatformListener("facebook", "post", blocking ? "blocking" : "nonblocking", callback);
-    }
-    if (events.includes("reshare") || events.includes("<all_events>")) {
-        registerPlatformListener("facebook", "reshare", blocking ? "blocking" : "nonblocking", callback);
-    }
-    if (events.includes("react") || events.includes("<all_events>")) {
-        registerPlatformListener("facebook", "react", blocking ? "blocking" : "nonblocking", callback);
-    }
-    if (events.includes("comment") || events.includes("<all_events>")) {
-        registerPlatformListener("facebook", "comment", blocking ? "blocking" : "nonblocking", callback);
-    }
-    fbPostContentInit();
+function notifyFilter(listener, listenerArguments, options) {
+    const reportedEvent = listenerArguments[0];
+    const ret = reportedEvent['platform'] == options['platform'] &&
+        options['eventTypes'].includes(reportedEvent['eventType']);
+    return ret;
 }
 
 /**
- * Register a callback for specific Reddit events. Supported events are "post", "comment",
- * "postVote", and "commentVote".
- * @param callback - the function to call when the event happens
- * @param [String] events - array of events to be tracked
- * @param blocking - whether the listener should be blocking. Allows canceling the event.
+ * A callback function for the social media activity event.
+ * @callback socialMediaActivityCallback
+ * @param {SocialMediaShareDetails} details - Additional information about the social media activity event.
  */
-export function registerRedditActivityTracker(
-    callback,
-    events,
-    blocking = false) {
-    if (events.includes("post") || events.includes("<all_events>")) {
-        registerPlatformListener("reddit", "post", blocking ? "blocking" : "nonblocking", callback);
-    }
-    if (events.includes("comment") || events.includes("<all_events>")) {
-        registerPlatformListener("reddit", "comment", blocking ? "blocking" : "nonblocking", callback);
-    }
-    if (events.includes("postVote") || events.includes("<all_events>")) {
-        registerPlatformListener("reddit", "postVote", blocking ? "blocking" : "nonblocking", callback);
-    }
-    if (events.includes("commentVote") || events.includes("<all_events>")) {
-        registerPlatformListener("reddit", "commentVote", blocking ? "blocking" : "nonblocking", callback);
-    }
-}
+
+/**
+ * Options when adding a social media activity event listener.
+ * @typedef {Object} socialMediaActivityOptions
```

Comment:
I've been trying to establish a convention of lower camel case for callback types and upper camel case for other types (including `@callback` functions that aren't really callbacks).